inbound.go

```diff
@@ -211,8 +211,18 @@ func (c *Connection) dispatchInbound(_ uint32, _ uint32, call *InboundCall, fram
 
 	// TODO(prashant): This is an expensive way to check for cancellation. Use a heap for timeouts.
 	go func() {
-		if <-call.mex.ctx.Done(); call.mex.ctx.Err() == context.DeadlineExceeded {
-			call.mex.inboundTimeout()
+		select {
+		case <-call.mex.ctx.Done():
+			if call.mex.ctx.Err() == context.DeadlineExceeded {
+				call.mex.inboundTimeout()
+			}
+		// If we see an error on the exchange, then we should exit immediately
+		case connErr := <-call.mex.errCh:
+			if c.log.Enabled(LogLevelDebug) {
+				c.log.Debugf("Got an error on the exchange: %v (unblock the application)", connErr)
+			}
+			call.Response().SendSystemError(connErr)
```

Comment:
if i don't send a response here, looks like the call seems to be still alive and is blocked (from the application side).. 
the test we have is based on the streaming branch where we have 2 processes server and client (not the per-process thing we discussed earlier today), crash the client and the server is expected to get an error but it doesn't (if i remove sending the response here)..